---
title: 'Project: Download and Begin Processing Data'
author: "Shaun"
date: "2024-02-25"
output:
  prettydoc::html_pretty:
    theme: cayman
    highlight: github
  toc: true

---

## TASK 1
# I may have the wrong genome. I have NW chrs. Find the genome without the patches 
# Run trim galore again with paired ends annotations


### Where did you get it from?
The transcriptomic data discussed here was deposited in the NCBI Gene Expression Omnibus (GEO) under accession number GSE65938, providing comprehensive access to all associated data files.

### What publication is it linked to?
The data originates from the study titled "RNA sequencing analysis of the developing chicken retina," published in Scientific Data in 2016.

### Who generated the data?
The experimental work was conducted as part of an ongoing project within the James Madison University’s Center for Genome & Metagenome Studies (CGEMS), aimed at elucidating transcriptional regulation in the developing vertebrate retina. The experiments were carried out with approval from the James Madison University Institutional Animal Care and Use Committee and adhered to the guidelines outlined in the National Institutes of Health guide for the care and use of laboratory animals.

### How was the RNA extracted?
RNA extraction was performed from eight embryonic chicken ocular tissues, including whole retinas from E8, E16, and E18 developing chicken embryos, as well as whole corneas collected from E18 embryos. Duplicates were obtained for each time point, and total RNA was extracted using a Qiagen AllPrep Mini Kit, following the manufacturer’s instructions. Isolated RNAs underwent an on-column DNaseI treatment step, were eluted in nuclease-free water, validated for quality and quantity using UV spectrophotometry, and stored long-term at -80°C. High-quality RNA samples were identified by an OD260/280 ratio falling between 1.9 and 2.1.

### What library prep was used?
The library preparation method employed was RNA sequencing (RNA-seq), a robust technique widely used for transcriptomic analysis.

### What cell type was used?
The focus of the study was on the retina and cornea of chicken embryos, rather than specific cell types within these tissues.

### What was the treatment/experimental condition?
The experimental conditions involved the collection of whole retinas and corneas from developing chicken embryos at different stages of embryonic development (E8, E16, and E18). The embryos were harvested and euthanized by decapitation, and total RNA was extracted from the ocular tissues.

### What sequencing platform was used?
Illumina Next Generation Sequencing (NGS) technology was utilized for sequencing the RNA samples, providing high-throughput and accurate transcriptomic data acquisition.



To download the data, I first saved all the accession numbers from GEO to a file using the SRA run selector, enabling me to run my script to download them. I obtained a total of 8 samples representing different embryonic timepoints of a chicken's retina and cornea.


```{bash, eval=FALSE}
(angsd) bash-4.4$ nano accession_list.txt
SRR1804235
SRR1804236
SRR1804237
SRR1804238
SRR1804239
SRR1804240
SRR1804241
SRR1804242
```

As my data is paired, I decided to split the reads on the forward and reverse strand into two separate files and then run FastQC on both.

```{bash, eval=FALSE}
(angsd) bash-4.4$ nano sra_download.sh 
(angsd) bash-4.4$ chmod u+x sra_download.sh 

#!/bin/bash
accession_num=$(cat accession_list.txt)


# make new directory to hold fastq files
#mkdir fastq
#cd fastq

# Read accession numbers from the file and iterate over them
while read -r accession_num; do
    echo "$accession_num"  # Print each accession number before downloading
    fasterq-dump "$accession_num" -F fastq -O fastq --skip-technical --split-3 --threads 3
done < accession_list.txt

echo "All downloads complete :)"

(angsd) bash-4.4$ bash sra_download.sh 
SRR1804235
spots read      : 39,324,028
reads read      : 78,648,056
reads written   : 78,648,056
SRR1804236
spots read      : 50,710,879
reads read      : 101,421,758
reads written   : 101,421,758
SRR1804237
spots read      : 72,207,041
reads read      : 144,414,082
reads written   : 144,414,082
SRR1804238
spots read      : 43,927,084
reads read      : 87,854,168
reads written   : 87,854,168
SRR1804239
spots read      : 33,887,041
reads read      : 67,774,082
reads written   : 67,774,082
SRR1804240
spots read      : 55,361,657
reads read      : 110,723,314
reads written   : 110,723,314
SRR1804241
spots read      : 28,727,301
reads read      : 57,454,602
reads written   : 57,454,602
SRR1804242
spots read      : 42,629,245
reads read      : 85,258,490
reads written   : 85,258,490

All downloads complete :)
```

Since I had 8 samples and split my reads into two separate files, I now have 16 .`fastq.gz files`, where '1' represents the forward strand and '2' represents the reverse strand.

```{bash, eval=FALSE}
[shp4022@cayuga-login1 fastq]$ ls | grep ".fastq.gz$"
SRR1804235_1.fastq.gz
SRR1804235_2.fastq.gz
SRR1804236_1.fastq.gz
SRR1804236_2.fastq.gz
SRR1804237_1.fastq.gz
SRR1804237_2.fastq.gz
SRR1804238_1.fastq.gz
SRR1804238_2.fastq.gz
SRR1804239_1.fastq.gz
SRR1804239_2.fastq.gz
SRR1804240_1.fastq.gz
SRR1804240_2.fastq.gz
SRR1804241_1.fastq.gz
SRR1804241_2.fastq.gz
SRR1804242_1.fastq.gz
SRR1804242_2.fastq.gz
[shp4022@cayuga-login1 fastq]$ ls | grep ".fastq.gz$" | wc -l
16
```


Next, I ran `FastQC` on my files. I inspected one sample, and it matched the results presented in the paper for `FastQC`.

```{bash, eval=FALSE}
(angsd) bash-4.4$ for file in *_*.fastq; do fastqc "$file"; done 
```

I then generated a `MultiQC` report.

```{bash, eval=FALSE}
(multiqc) [shp4022@cayuga-login1 fastq]$ multiqc *_fastqc.zip 
```

```{bash, eval=FALSE}
(base) shaunp@MAC306097 Classwork % scp -i ~/.ssh/Private_Key.txt shp4022@cayuga-login1.cac.cornell.edu:/athena/angsd/scratch/shp4022/project/fastq/multiqc_report.html  /Users/shaunp/Desktop/Weill_Cornell_Graduate/Grad_School/SPRING_2024/Analysis_Next-Gen_Sequencing_Data/Project/Fastqc/Project_multiqc_report.html 
multiqc_report.html                                                      100% 1448KB   7.5MB/s   00:00    
```


After examining the FastQC: adapter context for my reads, I trimmed my data.
# used the paired option 
```{bash, eval=FALSE}
(multiqc) [shp4022@cayuga-login1 fastq]$ mamba activate trim-galore
for file in *.fastq; do trim_galore --fastqc --stringency 3 $file; done
(multiqc) [shp4022@cayuga-login1 fastq]$ multiqc *_trimmed_fastqc.zip
```


```{bash, eval=FALSE}
(base) shaunp@MAC306097 Classwork % scp -i ~/.ssh/Private_Key.txt shp4022@cayuga-login1.cac.cornell.edu:/athena/angsd/scratch/shp4022/project/fastq/multiqc_report_trimmed.html  /Users/shaunp/Desktop/Weill_Cornell_Graduate/Grad_School/SPRING_2024/Analysis_Next-Gen_Sequencing_Data/Project/Fastqc/multiqc_report_trimmed.html
multiqc_report_trimmed.html                                   100% 1432KB   8.4MB/s   00:00    

```


<object data="./Fastqc/multiqc_report.html" type="text/html" width="100%" height="800px">
    <!-- Fallback content or error message if the browser doesn't support <object> -->
    Error: Unable to load the HTML file.
</object>


<object data="./Fastqc/multiqc_report_trimmed.html" type="text/html" width="100%" height="800px">
    <!-- Fallback content or error message if the browser doesn't support <object> -->
    Error: Unable to load the HTML file.
</object>


Next, I indexed my genome using the following script:

```{bash, eval=FALSE}
(angsd) [shp4022@cayuga-login1 genome]$ nano STAR_genomeGenerate.sh 
(angsd) [shp4022@cayuga-login1 genome]$ chmod u+x STAR_genomeGenerate.sh 
STAR --runMode genomeGenerate \
--runThreadN 5 \
--genomeDir galGal6_STAR_Index \
--genomeFastaFiles <(zcat galGal6.fa.gz) \
--sjdbGTFfile <(zcat galGal6.ncbiRefSeq.gtf.gz) \
--sjdbOverhang 99 \
--genomeSAindexNbases 10 
```

However, I encountered an error when trying to align one of the trimmed samples. Upon research, it appeared that trimming the data could alter the format and cause alignment errors. In my case, one of my reads, '@SRR1804237.72003183', could not align.

```{bash, eval=FALSE}
srun: job 115083 queued and waiting for resources
srun: job 115083 has been allocated resources
/athena/angsd/scratch/mef3005/share/envs/angsd/bin/STAR-avx2 --runMode alignReads --outFilterMultimapNmax 20 --runThreadN 8 --outFileNamePrefix E8_ --genomeDir ../genome/galGal6_STAR_Index --readFilesIn ../fastq/SRR1804237_1_trimmed.fq.gz ../fastq/SRR1804237_2_trimmed.fq.gz --readFilesCommand zcat --outFilterType BySJout --outSAMtype BAM SortedByCoordinate --outSAMattributes NH HI AS nM MD
STAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source
Feb 23 10:31:42 ..... started STAR run
Feb 23 10:31:42 ..... loading genome
Feb 23 10:31:50 ..... started mapping

EXITING because of FATAL ERROR in reads input: quality string length is not equal to sequence length
@SRR1804237.72003183
+
  .)<<AFFF..FFF7F)FF7F<FF
SOLUTION: fix your fastq file

Feb 23 11:21:55 ...... FATAL ERROR, exiting
srun: error: c0010: task 0: Exited with exit code 104
```


Therefore, I decided to use my untrimmed data to align to my genome, which is more consistent with the paper since they did not trim their data. Here is the full script for aligning all my samples.

## TASK 2

```{bash, eval=FALSE}
#!/bin/bash


# E8 retina rep 1 Align reads

STAR --runMode alignReads \
     --outFilterMultimapNmax 20 \
     --runThreadN 8 \
     --outFileNamePrefix E8_retina_rep1_ \
     --genomeDir ../genome/galGal6_STAR_Index \
     --readFilesIn ../fastq/SRR1804237_1.fastq.gz ../fastq/SRR1804237_2.fastq.gz \
     --readFilesCommand zcat \
     --outFilterType BySJout \
     --outSAMtype BAM SortedByCoordinate \
     --outSAMattributes NH HI AS nM MD


# E8 retina rep 2 Align reads

STAR --runMode alignReads \
--outFilterMultimapNmax 20 \
--runThreadN 8 \
--outFileNamePrefix E8_retina_rep2_ \
--genomeDir ../genome/galGal6_STAR_Index \
--readFilesIn ../fastq/SRR1804238_1.fastq.gz ../fastq/SRR1804238_2.fastq.gz \
--readFilesCommand zcat \
--outFilterType BySJout \
--outSAMtype BAM SortedByCoordinate \
--outSAMattributes NH HI AS nM MD

# E16 retina rep 1 Align reads

STAR --runMode alignReads \
--outFilterMultimapNmax 20 \
--runThreadN 8 \
--outFileNamePrefix E16_retina_rep1_ \
--genomeDir ../genome/galGal6_STAR_Index \
--readFilesIn ../fastq/SRR1804235_1.fastq.gz ../fastq/SRR1804235_2.fastq.gz \
--readFilesCommand zcat \
--outFilterType BySJout \
--outSAMtype BAM SortedByCoordinate \
--outSAMattributes NH HI AS nM MD

# E16 retina rep 2 Align reads

STAR --runMode alignReads \
--outFilterMultimapNmax 20 \
--runThreadN 8 \
--outFileNamePrefix E16_retina_rep2_ \
--genomeDir ../genome/galGal6_STAR_Index \
--readFilesIn ../fastq/SRR1804236_1.fastq.gz ../fastq/SRR1804236_2.fastq.gz \
--readFilesCommand zcat \
--outFilterType BySJout \
--outSAMtype BAM SortedByCoordinate \
--outSAMattributes NH HI AS nM MD

# E18 retina rep 1 Align reads

STAR --runMode alignReads \
--outFilterMultimapNmax 20 \
--runThreadN 8 \
--outFileNamePrefix E18_retina_rep1_ \
--genomeDir ../genome/galGal6_STAR_Index \
--readFilesIn ../fastq/SRR1804239_1.fastq.gz ../fastq/SRR1804239_2.fastq.gz \
--readFilesCommand zcat \
--outFilterType BySJout \
--outSAMtype BAM SortedByCoordinate \
--outSAMattributes NH HI AS nM MD


# E18 retina rep 2 Align reads

STAR --runMode alignReads \
--outFilterMultimapNmax 20 \
--runThreadN 8 \
--outFileNamePrefix E18_retina_rep2_ \
--genomeDir ../genome/galGal6_STAR_Index \
--readFilesIn ../fastq/SRR1804240_1.fastq.gz ../fastq/SRR1804240_2.fastq.gz \
--readFilesCommand zcat \
--outFilterType BySJout \
--outSAMtype BAM SortedByCoordinate \
--outSAMattributes NH HI AS nM MD


# E18 cornea rep 1 Align reads

STAR --runMode alignReads \
--outFilterMultimapNmax 20 \
--runThreadN 8 \
--outFileNamePrefix E18_cornea_rep1_ \
--genomeDir ../genome/galGal6_STAR_Index \
--readFilesIn ../fastq/SRR1804241_1.fastq.gz ../fastq/SRR1804241_2.fastq.gz \
--readFilesCommand zcat \
--outFilterType BySJout \
--outSAMtype BAM SortedByCoordinate \
--outSAMattributes NH HI AS nM MD

# E18 cornea rep 2 Align reads

STAR --runMode alignReads \
--outFilterMultimapNmax 20 \
--runThreadN 8 \
--outFileNamePrefix E18_cornea_rep2_ \
--genomeDir ../genome/galGal6_STAR_Index \
--readFilesIn ../fastq/SRR1804242_1.fastq.gz ../fastq/SRR1804242_2.fastq.gz \
--readFilesCommand zcat \
--outFilterType BySJout \
--outSAMtype BAM SortedByCoordinate \
--outSAMattributes NH HI AS nM MD

```

Confirmation that all 8 samples have been aligned using STAR:

```{bash, eval=FALSE}
[shp4022@cayuga-login1 Alignment]$ ls | grep "_Aligned.sortedByCoord.out.bam$"
E16_retina_rep1_Aligned.sortedByCoord.out.bam
E16_retina_rep2_Aligned.sortedByCoord.out.bam
E18_cornea_rep1_Aligned.sortedByCoord.out.bam
E18_cornea_rep2_Aligned.sortedByCoord.out.bam
E18_retina_rep1_Aligned.sortedByCoord.out.bam
E18_retina_rep2_Aligned.sortedByCoord.out.bam
E8_retina_rep1_Aligned.sortedByCoord.out.bam
E8_retina_rep2_Aligned.sortedByCoord.out.bam
[shp4022@cayuga-login1 Alignment]$ ls | grep "_Aligned.sortedByCoord.out.bam$" | wc -l
8
```


After alignment using `STAR`, I performed quality checks on my alignment for my reads using `qualimap bamqc`. To automate this process, I created the following script:

```{bash, eval=FALSE}
[shp4022@cayuga-login1 Alignment]$ nano align_qc.sh 
[shp4022@cayuga-login1 Alignment]$ chmod u+x align_qc.sh 
#!/bin/bash

for file in *.bam; do
        name=$(echo "$file" | sed "s/_Aligned.sortedByCoord.out.bam//")

        if [[ ! -f "$name.report.pdf" ]]; then
                qualimap bamqc -bam $file -outformat pdf -outfile "$name.report.pdf" -outdir /athena/angsd/scratch/shp4022/project/Alignment/Alignment_qc &
        fi
done

wait
```

I then downloaded the `bamqc` reports for my samples. Below I show one of those reports for one sample.

```{bash, eval=FALSE}
(base) shaunp@MAC306097 Classwork % scp -i ~/.ssh/Private_Key.txt shp4022@cayuga-login1.cac.cornell.edu:/athena/angsd/scratch/shp4022/project/Alignment/Alignment_qc/E16_retina_rep1.report.pdf   /Users/shaunp/Desktop/Weill_Cornell_Graduate/Grad_School/SPRING_2024/Analysis_Next-Gen_Sequencing_Data/Project/E16_retina_rep1.report.pdf 
E16_retina_rep1.report.pdf                                                                         100%  971KB   6.4MB/s   00:00    

```


<object data="./E16_retina_rep1.report.pdf" type="text/html" width="100%" height="800px">
    <!-- Fallback content or error message if the browser doesn't support <object> -->
    Error: Unable to load the HTML file.
</object>



















```{bash}
multiqc *Log.final.out* --outdir Alignment_qc
(multiqc) bash-4.4$ mv E8_Log.out E8_retina_rep1_Log.out
(multiqc) bash-4.4$ 
(multiqc) bash-4.4$ mv E8_Log.progress.out E8_retina_rep1_Log.progress.out
(multiqc) bash-4.4$ mv E8_Log.final.out E8_retina_rep1_Log.final.out
(multiqc) bash-4.4$ mv E8_SJ.out.tab E8_retina_rep1_SJ.out.tab
```





## FeatureCounts 

```{bash}
featureCounts -F gtf -g gene_id -T 5 -a ../genome/galGal6.ncbiRefSeq.gtf -o $name_output

sbatch -n1 --partition=angsd_class 
```




```{bash}
sbatch -n4 --partition=angsd_class --job-name=featureCounts fc.sh
(angsd) [shp4022@cayuga-login1 Alignment]$ sbatch -n4 --partition=angsd_class --job-name=featureCounts fc.sh
Submitted batch job 126767
squeue -u shp4022
```




